plugin/edit.py

```diff
@@ -45,7 +43,10 @@ def run(self, edit: Any, changes: Optional[List[TextEdit]] = None) -> None:
         if changes:
             last_row, last_col = self.view.rowcol(self.view.size())
             for change in reversed(sort_by_application_order(changes)):
-                start, end, newText = change
+                start, end, newText, version = change
+                if version is not None and version != self.view.change_count():
+                    debug('ignoring edit due to non-matching document version')
+                    continue
                 region = sublime.Region(self.view.text_point(*start), self.view.text_point(*end))
```

Comment:
While you're at it, these should be `text_point_utf16`.

Reply:
Ah, this is the master branch, never mind :)

Reply:
This is ST3 branch. :)